### PR TITLE
Upgrade ng2-bootstrap to rc.1

### PR DESCRIPTION
**Pull request by ciriarte**
Hi Guys, I've upgraded to ~~rc.0~~ rc.1, please let me know if it looks OK to you. These changes overlap with #472 but also include the new "@angular/*" references.

### CODE REVIEW

File: components/typeahead/typeahead-container.component.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ const TEMPLATE:any = {
       style="display: block"
       [ngStyle]="{top: top, left: left, display: display}"
       (mouseleave)="focusLost()">
-    <li *ngFor="#match of matches"
+    <li *ngFor=let match of matches"
```
Comment (francesconi) [inline components/typeahead/typeahead-container.component.ts]: It seems you're missing a quote `<li *ngFor="let match of matches"`

**Comment by hhsadiq** [comment]
We need the upgrade to rc0 urgently. There are many breaking changes and previous version of ng2-bootstrap is not supporting rc0..Thanks alot

File: components/typeahead/typeahead-container.component.ts (hunk 1 above)
Comment (ciriarte) [inline components/typeahead/typeahead-container.component.ts]: Thanks! I've added the missing quote.

**Comment by coutin** [comment]
You could probably push to Rc1 :-) 

Thx for the job done.

**Comment by ciriarte** [comment]
@coutin working on that!

**Comment by laurelnaiad** [comment]
It's somewhat troubling and now a major blocker that ng2-bootstrap is still on beta 15. What can we in the peanut gallery do to get it up to date? Is there any feedback the maintainers could provide on this PR that would help us make it happen? I'm sure there are quite a few of us who would be willing to help with whatever is holding it up. Cheers!

**Comment by ciriarte** [comment]
@valorkin sorry to bother you, any chance you can help us with this PR?

**Comment by rodolfocop** [comment]
I am willing to help in any need. I need and that this library is in RC.1 version

**Comment by valorkin** [comment]
Hey guys, sorry for being absent, I am at ngconf now.

**Comment by valorkin** [comment]
Huge thanks for your help! Let me try locally, if all good I will publish it in a sec!

**Comment by laurelnaiad** [comment]
I've pushed a slightly modified version of @ciriarte's PR **with the built files included** to my github fork so that the branch can be used from npm (and jspm).

https://github.com/laurelnaiad/ng2-bootstrap/tree/rc0-compat-temp

Basically, I just backed off of rc.1 in favor of rc.0, since I don't think there's a compelling reason to force use of rc.1, and I tweaked some other dependencies, but the main feature is that it has the built files in the branch.

**Comment by hongbo-miao** [comment]
@valorkin glad to see you are back to ng2-bootstrap!

**Comment by laurelnaiad** [comment]
Happy to see we cross-posted. :)

**Comment by valorkin** [comment]
@Hongbo-Miao sorry, I will miss a couple of days more
but when I will get back from ng-conf
things go faster :)

**Comment by rodolfocop** [comment]
@valorkin thanks man for attention, I can now use the ng2-file-upload with ng2-bootstrap

**Comment by theunreal** [comment]
Is it just me, or you get the following errors with this merge?

```
node_modules/ng2-bootstrap/components/alert/alert.component.ts(25,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<AlertComponent>'.
  Type '{}' is not assignable to type 'AlertComponent'.
    Property 'type' is missing in type '{}'.
node_modules/ng2-bootstrap/components/common.d.ts(1,34): error TS2307: Cannot find module 'angular2/core'.
node_modules/ng2-bootstrap/components/datepicker/datepicker-inner.component.ts(91,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<Date>'.
  Type '{}' is not assignable to type 'Date'.
    Property 'toDateString' is missing in type '{}'.
node_modules/ng2-bootstrap/components/dropdown/dropdown.directive.ts(20,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<boolean>'.
  Type '{}' is not assignable to type 'boolean'.
node_modules/ng2-bootstrap/components/dropdown/dropdown.directive.ts(21,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<boolean>'.
  Type '{}' is not assignable to type 'boolean'.
node_modules/ng2-bootstrap/components/pagination/pagination.component.ts(95,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<number>'.
  Type '{}' is not assignable to type 'number'.
node_modules/ng2-bootstrap/components/pagination/pagination.component.ts(96,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<PageChangedEvent>'.
  Type '{}' is not assignable to type 'PageChangedEvent'.
    Property 'itemsPerPage' is missing in type '{}'.
node_modules/ng2-bootstrap/components/rating/rating.component.ts(33,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<number>'.
  Type '{}' is not assignable to type 'number'.
node_modules/ng2-bootstrap/components/rating/rating.component.ts(34,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<number>'.
  Type '{}' is not assignable to type 'number'.
node_modules/ng2-bootstrap/components/tabs/tab.directive.ts(21,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<TabDirective>'.
  Type '{}' is not assignable to type 'TabDirective'.
    Property 'heading' is missing in type '{}'.
node_modules/ng2-bootstrap/components/tabs/tab.directive.ts(22,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<TabDirective>'.
  Type '{}' is not assignable to type 'TabDirective'.
node_modules/ng2-bootstrap/components/tabs/tab.directive.ts(23,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<TabDirective>'.
  Type '{}' is not assignable to type 'TabDirective'.
node_modules/ng2-bootstrap/components/typeahead/typeahead.directive.ts(24,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<boolean>'.
  Type '{}' is not assignable to type 'boolean'.
node_modules/ng2-bootstrap/components/typeahead/typeahead.directive.ts(25,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<boolean>'.
  Type '{}' is not assignable to type 'boolean'.
node_modules/ng2-bootstrap/components/typeahead/typeahead.directive.ts(26,3): error TS2322: Type 'EventEmitter<{}>' is not assignable to type 'EventEmitter<{ item: any; }>'.
  Type '{}' is not assignable to type '{ item: any; }'.
    Property 'item' is missing in type '{}'.
```

**Comment by laurelnaiad** [comment]
>    "rxjs": "5.0.0-beta.2" 

is incorrect. It should be pointing at beta 6 at minimum.

This was one ot the reasons I adjusted the deps in my branch above (https://github.com/laurelnaiad/ng2-bootstrap/commit/8db45570e724cd8806828671fed524b67f9f6a79)

**Comment by ps2goat** [comment]
@theunreal - EventEmitters now need type specifications when being initialized.

E.g.:

`let x: EventEmitter<string> = new EventEmitter<string>();`

This was previously done like this:
`let x: EventEmitter<string> = new EventEmitter();`

btw, this error does not show up in Visual Studio Update 2, only when I compile with gulp-typescript and typescript 1.8.10.  We're coming off of 1.7.5, though, so I don't know about anything in-between.

**Comment by ps2goat** [comment]
+1 for RC 1 support

**Comment by valorkin** [comment]
Thanks to all of you guys!
Just published

**Comment by valorkin** [comment]
Can someone PR all contributors manifest support? :)

**Pull request merged**